packages/rpc/lib/src/utils/unix_rpc_client.dart

```diff
@@ -17,12 +18,18 @@ class UnixRPCClient extends ServerProxyBase {
     var socket = await Socket.connect(address, 0);
     socket.add(utf8.encode(package));
 
-    socket.listen((event) {
-      LogManager.getInstance.debug('Event received is ${utf8.decode(event)}');
-      completer.complete(utf8.decode(event));
-    },
+    socket.listen(
+        (event) {
+          LogManager.getInstance
+              .debug('Event received is ${utf8.decode(event)}');
+          completer.complete(utf8.decode(event));
+        },
         onDone: () => LogManager.getInstance.debug('End JSON RPC Stream'),
-        onError: (err) => LogManager.getInstance.error(err));
+        onError: (error) {
+          LogManager.getInstance.error(error);
+          throw LNClientException(
+              error["code"], error["message"], error["data"]);
+        });
```

Comment:
Ok this is my mistake! the onError callback is trigger for other reason  maybe related to the stream.

In the CI i noted that there is the following error

```
cln_1       | ┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
cln_1       | │ #0   LogManager.info (package:cln_common/src/logger/logger_manager.dart:24:13)
cln_1       | │ #1   RPCClient.connect (package:clightning_rpc/src/clightning_rpc.dart:10:10)
cln_1       | ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
cln_1       | │ 💡 Connection to JSON RPC 2.0 at the following path /workdir/lightning_dir_one/regtest/lightning-rpc
cln_1       | └───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
cln_1       | ┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
cln_1       | │ #0   LogManager.debug (package:cln_common/src/logger/logger_manager.dart:16:13)
cln_1       | │ #1   UnixRPCClient.transmit.<anonymous closure> (package:clightning_rpc/src/utils/unix_rpc_client.dart:21:30)
cln_1       | ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
cln_1       | │ 🐛 Event received is {"jsonrpc":"2.0","result":{"foo_opt":"hello"},"id":791291083}
cln_1       | │ 🐛 
cln_1       | │ 🐛 
cln_1       | └───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

And I think we need the fix looks like this (please do no apply I do not test the code in the local so I do not know if this is correct dart)

```dart
 socket.listen(
        (event) {
          LogManager.getInstance
              .debug('Event received is ${utf8.decode(event)}');
         if (event.containsKey("errors")) {
              var error = event["errors"];
            var error = LNClientException(
              error["code"], error["message"], error["data"]);
            completer.completeException(error);
           return;
         }
          completer.complete(utf8.decode(event));
        },
        onDone: () => LogManager.getInstance.debug('End JSON RPC Stream'),
        onError: (error) {
          LogManager.getInstance.error(error);
          completer.completeException(error);
        });
```